src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/google/HttpGoogleDataprocDAO.scala

```diff
@@ -0,0 +1,417 @@
+import com.google.api.client.http.HttpResponseException
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.services.bigquery.BigqueryScopes
+import com.google.api.services.cloudresourcemanager.CloudResourceManager
+import com.google.api.services.compute.model.Firewall.Allowed
+import com.google.api.services.compute.model.{Firewall, Instance}
+import com.google.api.services.compute.{Compute, ComputeScopes}
+import com.google.api.services.dataproc.Dataproc
+import com.google.api.services.dataproc.model.{Cluster => DataprocCluster, ClusterConfig => DataprocClusterConfig, ClusterStatus => DataprocClusterStatus, Operation => DataprocOperation, _}
+import com.google.api.services.oauth2.{Oauth2, Oauth2Scopes}
+import org.broadinstitute.dsde.workbench.google.GoogleCredentialModes.GoogleCredentialMode
+import org.broadinstitute.dsde.workbench.google.AbstractHttpGoogleDAO
+import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterStatus.ClusterStatus
+import org.broadinstitute.dsde.workbench.leonardo.model.google._
+import org.broadinstitute.dsde.workbench.metrics.GoogleInstrumentedService
+import org.broadinstitute.dsde.workbench.metrics.GoogleInstrumentedService.GoogleInstrumentedService
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GcsPath, GoogleProject}
+import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail, WorkbenchUserId}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class HttpGoogleDataprocDAO(appName: String,
+                            googleCredentialMode: GoogleCredentialMode,
+                            override val workbenchMetricBaseName: String,
+                            defaultNetworkTag: NetworkTag,
+                            defaultRegion: String)
+                           (implicit override val system: ActorSystem, override val executionContext: ExecutionContext)
+  extends AbstractHttpGoogleDAO(appName, googleCredentialMode, workbenchMetricBaseName) with GoogleDataprocDAO {
+
+  override implicit val service: GoogleInstrumentedService = GoogleInstrumentedService.Dataproc
+
+  override val scopes: Seq[String] = Seq(ComputeScopes.CLOUD_PLATFORM)
+
+  private lazy val oauth2Scopes = List(Oauth2Scopes.USERINFO_EMAIL, Oauth2Scopes.USERINFO_PROFILE)
+  private lazy val bigqueryScopes = List(BigqueryScopes.BIGQUERY)
+
+  private val httpTransport = GoogleNetHttpTransport.newTrustedTransport
+  private val jsonFactory = JacksonFactory.getDefaultInstance
+
+  private lazy val dataproc = {
+    new Dataproc.Builder(httpTransport, jsonFactory, googleCredential)
+      .setApplicationName(appName).build()
+  }
+
+  private lazy val compute = {
+    new Compute.Builder(httpTransport, jsonFactory, googleCredential)
+      .setApplicationName(appName).build()
+  }
+
+  // TODO move out of this DAO
+  private lazy val oauth2 =
```

Comment:
I'd like to move oauth and project-y things out of this DAO as well, but it's just a couple things not likely to change -- can do this later.